### ISSUE TITLE
[BUDI-7595] Date only external date field seems to be affected by time zoning.

### ISSUE THREAD

**Issue by ConorWebb96** [Budibase/budibase#11977]
**Checklist**
- [x] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**
- Self
  - Method: docker compose
  - Budibase 2.11.8
  - App Version: 2.11.8

**Describe the bug**
Time zoning is still affecting the newly created date only fields with the toggle `Date only` toggled on. Basically any date that gets selected is put back 1 day depending on which timezone you are in. This happens both in the builder and through the forms within the design area.

**To Reproduce**
1. Set your machine time to CDT - Chicago, IL - United States
2. Set your database to GMT (Not sure this will have any affect)
3. Use the Schema below to create your Postgres tables
4. Add the postgres data source to Budibase
5. Fetch the newly created table.
6. Add a record filling in the date only field.
7. See that the day goes back 1 day. Shows up correctly in the database.

**Expected behavior**
Date saved to the database should be whats displayed within Budibase itself.

**Screenshots**
![Screen Recording 2023-10-05 at 04 08 40](https://github.com/Budibase/budibase/assets/126772285/bcdef16b-ad62-4ee3-a77e-cd0bd8405d50)

**App Export**
[timezone-export-1696497214047.tar.gz](https://github.com/Budibase/budibase/files/12816525/timezone-export-1696497214047.tar.gz)

**Schema Export**
```sql
CREATE TABLE public.timezone (
    id integer NOT NULL,
    date_field date,
    datetime_field timestamp without time zone,
    name_field text
);
```
**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Brave
 - Version 1.58.135

<sub>[BUDI-7595](https://linear.app/budibase/issue/BUDI-7595/date-only-external-date-field-seems-to-be-affected-by-time-zoning)</sub>

**Comment by leolivier** [comment]
I have this issue also with internal budibase DB. I created a field named birthday with the flag dateonly but I saw when I tried to format the date taht some of them where 1 day before the one I entered. Looking in a dynamic field to the {{birthdate}} value, the date still contains a time : 
![image](https://github.com/Budibase/budibase/assets/4014752/fc0685a3-6c12-49d5-aa97-46f2f1b48bcc)
whereas the displayed date is : 
![image](https://github.com/Budibase/budibase/assets/4014752/0bb6d030-1a02-4ee7-8ff4-b2f024bc41c1)

Furthermore, whatever the value of the flag dateonly, the datepicker always asks for a time!
